protovalidate/internal/extra_func.py

```diff
@@ -129,23 +106,27 @@ def is_ip_prefix(val: celtypes.Value, *args) -> celpy.Result:
     elif len(args) == 2 and (not isinstance(args[0], celtypes.IntType) or not isinstance(args[1], celtypes.BoolType)):
         msg = "invalid argument, expected int and bool"
         raise celpy.CELEvalError(msg)
-    try:
-        if version is None:
-            ip_network(val, strict=bool(strict))
-        elif version == 4:
-            IPv4Network(val, strict=bool(strict))
-        elif version == 6:
-            IPv6Network(val, strict=bool(strict))
-        else:
-            msg = "invalid argument, expected 4 or 6"
-            raise celpy.CELEvalError(msg)
-        return celtypes.BoolType(True)
-    except ValueError:
-        return celtypes.BoolType(False)
+
+    return celtypes.BoolType(_is_ip_prefix(val, version, strict=strict))
+
+
+def _is_ip_prefix(string: str, version: int, *, strict=False) -> bool:
```

Comment:
Not sure of the best way to handle this. Other validation methods call this internally and if we stick with just `is_ip_prefix` then they have to pass cel values and handle cel types.

We have a similar scenario with Java (and C++). We could go back to the `is_email` and `validate_email` pattern that was in use before. 🤷

Reply:
Yes, looks like a good reason for a separate function to me.

`is_` vs. `validate_` is not self-explanatory. What if we prefix the type converting function with `cel_`?

```python
def cel_is_ip(val: celtypes.Value, ver: typing.Optional[celtypes.Value] = None) -> celpy.Result:
   ...
   return celtypes.BoolType(is_ip(...))

def is_ip(val: str, ver: int) -> bool:
   ...
```

Reply:
Updated. I also updated the other functions that don't require an internal implementation just to keep things consistent.